kornia/core/data/tensor_base.py

```diff
@@ -0,0 +1,25 @@
+from typing import List, Union
+from torch import Tensor
+
+
+class TensorBase(Tensor):
+
+    @staticmethod
+    def __new__(cls, tensor: Tensor, *args, **kwargs):
+        return super().__new__(cls, tensor, *args, **kwargs)
+
+    def __init__(
+        self, tensor: Tensor
+    ) -> None:
+        pass
+
+    def transform(self, mat: Tensor):
```

Comment:
Not convinced about this name, a transform can be geometric, colorspaces, just illumination, etc /cc @ducha-aiki  @anguelos @lferraz